internal_load_balancer.tf

```diff
@@ -0,0 +1,121 @@
+data "google_compute_network" "default" {
+  name = var.network
+}
+
+resource "google_compute_subnetwork" "elastic_lb_subnet" {
+  name          = "es-load-balancer-subnetwork"
+  ip_cidr_range = "192.168.254.0/24"
+  region        = var.region
+  network       = data.google_compute_network.default.self_link
+}
+
+resource "google_compute_region_target_http_proxy" "elasticsearch" {
+  provider = google-beta
+
+  region  = var.region
+  name    = "es-backend-proxy"
+  url_map = google_compute_region_url_map.default.id
+}
+
+resource "google_compute_region_url_map" "default" {
+  provider = google-beta
+
+  region   = var.region
+  # url map name is displayed as load balancer name, this name maybe misleading
+  name            = "es-ilb"
+  default_service = google_compute_region_backend_service.elasticsearch.id
+}
+
+
+resource "google_compute_forwarding_rule" "elasticsearch" {
+  provider = google-beta
+
+  all_ports              = false
+  allow_global_access    = false
+  is_mirroring_collector = false
+
+  name = "elasticsearch-forwarding-rule"
+
+  network    = data.google_compute_network.default.self_link
+  port_range = "80-80"
+
+  load_balancing_scheme = "INTERNAL_MANAGED"
+  ip_protocol           = "TCP"
+
+  region = var.region
+  target = google_compute_region_target_http_proxy.elasticsearch.self_link
+
+  # used as DNS reference in internal GCE DNS system, service without label do not get DNS name
+  service_label = "es-ilb"
+}
+
+resource "google_compute_instance_group" "elasticsearch" {
+  provider    = google-beta
+  name        = "elasticsearch-instance-pool-${data.google_compute_zones.available.names[count.index]}"
+  description = "Elasticsearch instance pool ${data.google_compute_zones.available.names[count.index]}"
+  zone        = var.zone != null ? var.zone : data.google_compute_zones.available.names[count.index]
+  count       = var.zone != null ? 1 : var.node_count < local.zone_count ? var.node_count : local.zone_count
+
+  instances = [
+    for i in google_compute_instance.elasticsearch : i.self_link if
```

Comment:
This seems quite complex and I am not sure I understand it. Can you please explain or add some comment?

Reply:
How about now?